web/wp-config.php

```diff
@@ -27,11 +27,11 @@
 	) )->notEmpty();
 }
 
-/**
- * Disallow on server file edits
- */
-define( 'DISALLOW_FILE_EDIT', true );
-define( 'DISALLOW_FILE_MODS', true );
+// FS writes aren't permitted in test or live, so we should let WordPress know to disable relevant UI
+if (in_array($_ENV['PANTHEON_ENVIRONMENT'], array( 'test', 'live' )) && ! defined('DISALLOW_FILE_MODS')) {
+    define( 'DISALLOW_FILE_EDIT', true );
```

Comment:
What is DISALLOW_FILE_EDIT? Defining it without a `!defined()` guard will prevent the end user from modifying it. Also, you are comparing wp-config.php here with wp-config-pantheon.php in our WordPress upstream. wp-config.php is the file the user takes ownership of and may freely edit, whereas wp-config-pantheon.php is owned and updated by Pantheon. Seems we should be consistent in structure here compared to the non-Composer WP upstream, just as we are with Drupal.

Reply:
I started writing a suggested change where we add a `!defined` guard but then realized the pre-patched version already adds it unguarded, so thought it weren't broke.

Reply:
The pre-patched version comes straight from WordPress, right? The intention in WordPress is that wp-config.php is adopted by the user, so guarding the defines is not required. If we move this code to wp-config-pantheon.php, then we will need to guard it.

Reply:
Re: the structure should we intentionally go a different route from https://github.com/pantheon-systems/example-wordpress-composer/tree/master/web, or does your point apply over there too?

Reply:
That project was made by Andrew Taylor before wp-config-pantheon.php was split out of the Pantheon WordPress upstream. It was never brought up-to-date. I think our new Composer-based repo should be.

Reply:
@greg-1-anderson: re `DISALLOW_FILE_EDIT`: WordPress, by default (I think?), allows you the ability edit theme and plugin files directly from within the WordPress wp-admin. This setting just disables that. We should probably make this a default since it's a pretty bad pattern in general but a guarded approach could work for now.

I think I get what the suggestion is but am hesitant to do a bigger refactor of `wp-config.php`. I did look at it in a bit more detail and saw that we have 3 different checks of `if ( isset( $_ENV['PANTHEON_ENVIRONMENT'] ) )`, each doing different things. I consolidated those into one and moved some code around.

Re: `wp-config-pantheon.php` I'm having doubts because it's confusing:
* we currently define the DB_NAME, DB_USER, etc. in the local `wp-config.php` and then it gets blury. I think we have a separate repo (https://github.com/pantheon-systems/wordpress-composer) from where we fetch `wp-config-pantheon.php` but I don't see it there. Where is it coming from and how can I edit it?
* I see it on my local once I run `composer install` under `web/wp/wp-config-pantheon.php` but there's no require or include in the local `wp-config.php` similar to https://github.com/pantheon-systems/WordPress/blob/default/wp-config.php#L14-L15. I'm not even sure it gets taken into account but I assume it's not.

If I'm confused, will our customers be as well? To simplify things, could we not just have a local `wp-config-pantheon.php` next to `wp-config.php`?

Reply:
Since this repo is an upstream, yes, you could have a local [wp-config-pantheon.php](https://github.com/pantheon-systems/WordPress/blob/default/wp-config-pantheon.php) right in this repository. I would use the files in pantheon-system/WordPress as a starting point. Definitely get rid of the `else:` syntax.

Reply:
@greg-1-anderson want to have another look?